### PR TITLE
Remove checked-in symlinks

### PR DESCRIPTION
**Pull request by wolfs**
These caused cache misses between Windows and Linux

### CODE REVIEW

File: subprojects/core/src/test/groovy/org/gradle/api/file/FileCollectionSymlinkTest.groovy [hunk 1]
```diff
@@ -15,24 +15,34 @@
...
+        symlink.createLink(symlinked)
+    }
 
-    @Shared File baseDir = getTestFile("symlinks")
-    @Shared File file = new File(baseDir, "file")
-    @Shared File symlink = new File(baseDir, "symlink")
-    @Shared File symlinked = new File(baseDir, "symlinked")
+    @Unroll
+    def "project.#method can handle symlinks"() {
+        def fileCollection = this."${method}"()
```
Comment (lptr) [inline subprojects/core/src/test/groovy/org/gradle/api/file/FileCollectionSymlinkTest.groovy]: I think adding `baseDir` as a parameter here would clarify why we have some files in this file-collection.

Comment (wolfs) [reply subprojects/core/src/test/groovy/org/gradle/api/file/FileCollectionSymlinkTest.groovy]: I did that. I am not sure it is better now. WDYT?

Comment (lptr) [reply subprojects/core/src/test/groovy/org/gradle/api/file/FileCollectionSymlinkTest.groovy]: What I meant is to change this:

    def fileCollection = this."${method}"()

to this:

    def fileCollection = this."${method}"(baseDir)

Comment (wolfs) [reply subprojects/core/src/test/groovy/org/gradle/api/file/FileCollectionSymlinkTest.groovy]: Yeah, but then `files` will have an used parameter.

Comment (lptr) [reply subprojects/core/src/test/groovy/org/gradle/api/file/FileCollectionSymlinkTest.groovy]: Ah, I see what you did there... I missed that you were calling the method on `this` and not on `project`... I think this is unnecessary magic, and would prefer the previous solution where the code to execute was clearly expressed in the `where:` clause.

Comment (wolfs) [reply subprojects/core/src/test/groovy/org/gradle/api/file/FileCollectionSymlinkTest.groovy]: Better now?

**Pull request merged**